---
title: "BIOL4315_Lab2_Report"
output: html_document
date: "2025-09-16"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Long Read Quality Control

(sequali-env)$conda install filtlong

(sequali-env)$filtlong --keep_percent 75 ERR1539001.fastq.gz | gzip > ERR1539001-filtlong.fastq.gz

(sequali-env)$sequali --outdir ./sequali_reports_filtlong ERR1539001-filtlong.fastq.gz

After the reads were filtered: N50 increased from 6681 bp to 7371 bp, mean read length increased from 4888.03 bp to 6914.58, distribution of read lengths changed from skewed left to normal, average quality score per read 10 or greater increased from 2.44% to 4.49%, and per position quality score had a much smaller drop off after 10000 bp.

The quality of the data improved after filtering.  However, the quality of this sequencing run may not be adequate for downstream analysis.  The proportion of reads with an average quality score 10 or greater is too low (4.49%). To use data for downstream analysis, we would like this proportion to be greater than 50%.


## Short Read Quality Control

(base)$wget -nc ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR112/040/ERR11203340/ERR11203340_1.fastq.gz

(base)$wget -nc ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR112/040/ERR11203340/ERR11203340_2.fastq.gz

```{r Campilobacter jejuni}
library(Rqc)
qcRes <- rqc(path = "./c_jejuni/", pattern = ".fastq.gz", openBrowser = FALSE)
```
```{r fig.cap = "The base call quality drops below twenty at position 101 in the first run."}
rqcCycleQualityBoxPlot(qcRes)
```

```{r fig.cap = "N is not found in significant proportions for any of the base calls."}
rqcCycleBaseCallsLinePlot(qcRes)
```

```{r fig.cap = "Just under half of the sequences have one read.  Almost three quarters of the sequences have one or two reads." }
rqcReadFrequencyPlot(qcRes)
```

```{r QuasR}
library(QuasR)
```

```{r Trim}
fastqFiles <- c("./c_jejuni/ERR11203340_1.fastq.gz", "./c_jejuni/ERR11203340_2.fastq.gz")
outFiles <- paste(pattern = c("processed_1_", "processed_2_"), ".fastq", sep = "")
preprocessReads(fastqFiles, outFiles, truncateEndBases = 1)
```

```{r ShortRead}
library(ShortRead)
```

```{r TrimCorrect}
reads <- readFastq("./c_jejuni/ERR11203340_1.fastq.gz")
trimmedReads <- trimTails(reads, k = 1, a = 10, left = FALSE, right = TRUE, successive = TRUE)
writeFastq(trimmedReads, "trimmed_reads.fastq.gz", compress = TRUE)
```